kernel/src/expressions/literal_expression_transform.rs

```diff
@@ -132,7 +132,11 @@ impl<'a, T: Iterator<Item = &'a Scalar>> SchemaTransform<'a> for LiteralExpressi
         let field_exprs = self.stack.split_off(mark);
 
         if field_exprs.len() != struct_type.fields_len() {
-            self.set_error(Error::InsufficientScalars);
+            // Only set InsufficientScalars error if there isn't already an error set
+            // (e.g., a type mismatch error that prevented expressions from being created)
+            if self.error.is_ok() {
+                self.set_error(Error::InsufficientScalars);
+            }
```

Comment:
Ah good catch! So somewhere while recursing into the struct we hit an error for type mismatch and set the error but don't do the 'return early' then? I guess is the issue that we _do_ return early but it's from within `recurse_into_struct`?

I wonder if we should modify `self.set_error` to perform this check and only set the error if it doesn't yet exist?